### DISCUSSION TITLE
Example on how to pass packages from flakes into home.packages

### DISCUSSION THREAD

**Discussion by SuperSandro2000** [nix-community/home-manager#7456] [Q&A]
I am using flakes with home-manager and want to add a packages from some flake into `home.packages`. Currently I hacked something together and passing in some things via `extraSpecialArgs` and then use them as input.
My question is now: How are you really supposed to do that platform independent? I also couldn't figure out a way to use [eachSystem](https://github.com/numtide/flake-utils#eachsystem---system---system---attrs) since the output is not what home-manager is expecting.

**Comment by berbiche** [comment]
Hi,

If you have access to `pkgs` in your expression, then you can write it as `my-flake.packages.${pkgs.stdenv.hostPlatform.system}` or `pkgs.system`.
This is the only way that I know to use flake's "packages" output.

**Comment by SuperSandro2000** [comment]
> I am using flakes with home-manager and want to add a packages from some flake into `home.packages`. Currently I hacked something together and passing in some things via `extraSpecialArgs` and then use them as input.

I ended up doing an overlay which seemed to be the best option.

```nix
configuration = _: {
  imports = [ ./home.nix ];
  nixpkgs.overlays = [
    (final: prev: {
    nix-output-monitor = nix-output-monitor.defaultPackage."${system}";
    })
  ];
};
```

Still, I am wondering if this is the best solution, or if there is a better one.

**Comment by terlar** [comment]
For whatever it is worth I’m using the overlay option as well. Also interested to know if there is an even better one.

**Comment by teto** [comment]
what I do is pass the flake `self` as `extraSpecialArgs`

```
          home-manager.extraSpecialArgs = {
            secrets = lib.optionalAttrs withSecrets secrets;
            inherit withSecrets;
            flakeSelf = self;
            inherit dotfilesPath secretsFolder;
          };
```
and then I can e.g., add 
`    flakeSelf.inputs.rest-nvim.packages.${pkgs.stdenv.hostPlatform.system}.rest-nvim-dev`
as a package

**Reply by teto** [reply]
one can also set `home-manager.useGlobalPkgs = true` to have nixpkgs' overlays apply to home-manager without the need to reapply the overlay
